src/telliot_feeds/reporters/tips/listener/utils.py

```diff
@@ -0,0 +1,62 @@
+from collections import defaultdict
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+
+def filter_batch_result(data: Dict[Any, Any]) -> defaultdict[Any, List[Any]]:
+    """Filter data dictionary with tuple key
+
+    >>> example {(current_value, queryid): 0, (current_timestamp, queryid): 123}
+    to {queryid: [0, 123]}
+
+    Args:
+    - data: dictionary
+
+    Return:
+    - dictionary with value of list type
+    """
+    mapping = defaultdict(list)
+    results = defaultdict(list)
+
+    for k in data:
+        mapping[k[1]].append(k)
+    for query_id in mapping:
+        for val in mapping[query_id]:
+            value = data[val]
+            results[query_id].append(value)
+
+    return results
+
+
+def sum_values(x: Optional[int], y: Optional[int]) -> Optional[int]:
+    """Takes two values and returns sum and handles Nonetype"""
+    return sum((num for num in (x, y) if num is not None))
+
+
+def sort_by_max_tip(dict: Dict[bytes, int]) -> list[Tuple[bytes, int]]:
+    """Takes dictionary of int type value and sorts by max value"""
+    sorted_lis = sorted(dict.items(), key=lambda item: item[1], reverse=True)
```

Comment:
You can just return the sorted list, you don't have to create a new var if only used once like this